src/org/gk/qualityCheck/QACheckUtilities.java

```diff
@@ -26,4 +33,297 @@ public static GKInstance getLatestCuratorIEFromInstance(GKInstance instance) thr
+                physicalEntities.addAll(findAllConstituentPEs(physicalEntity));
+            // If EWAS, just adds to Set and is returned.
+            } else if (physicalEntity.getSchemClass().isa(ReactomeJavaConstants.EntityWithAccessionedSequence)) {
+                physicalEntities.add(physicalEntity);
+            }
+        }
+        return physicalEntities;
+    }
+
+    /**
+     * Checks if PhysicalEntity contains additional PhysicalEntities within it.
+     * @param physicalEntity GKInstance -- PhysicalEntity that is being checked.
+     * @return boolean -- true if PhysicalEntity type that contains multiple PEs, false if only type that has single PhysicalEntity.
+     */
+    public static boolean containsMultiplePEs(GKInstance physicalEntity) {
+        if (physicalEntity.getSchemClass().isa(ReactomeJavaConstants.Complex) ||
+                physicalEntity.getSchemClass().isa(ReactomeJavaConstants.Polymer) ||
+                physicalEntity.getSchemClass().isa(ReactomeJavaConstants.EntitySet)) {
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Finds all PhysicalEntities contained within a Complex, Polymer or EntitySet. Searches recursively by calling parent 'findAllPhysicalEntities' method.
+     * @param multiPEInstance GKInstance -- Complex, Polymer or EntitySet instance that will be searched for all PhysicalEntity instances.
+     * @return Set<GKInstance> -- All distinct PhysicalEntities that are found in Complex/Polymer/EntitySet.
+     * @throws Exception -- Thrown by MySQLAdaptor.
+     */
+    public static Set<GKInstance> findAllConstituentPEs(GKInstance multiPEInstance) throws Exception {
+        Set<GKInstance> physicalEntities = new HashSet<>();
+        if (multiPEInstance.getSchemClass().isa(ReactomeJavaConstants.Complex)) {
+            // Recursively search Complex components for all PhysicalEntities.
+            Collection<GKInstance> complexComponents = multiPEInstance.getAttributeValuesList(ReactomeJavaConstants.hasComponent);
+            for (GKInstance complexComponent : complexComponents) {
+                physicalEntities.addAll(findAllPhysicalEntities(complexComponent));
+            }
+        } else if (multiPEInstance.getSchemClass().isa(ReactomeJavaConstants.Polymer)) {
+            // Recursively search Polymer repeatedUnits for all PhysicalEntities.
+            Collection<GKInstance> polymerUnits = multiPEInstance.getAttributeValuesList(ReactomeJavaConstants.repeatedUnit);
+            for (GKInstance polymerUnit : polymerUnits) {
+                physicalEntities.addAll(findAllPhysicalEntities(polymerUnit));
+            }
+        } else if (multiPEInstance.getSchemClass().isa(ReactomeJavaConstants.EntitySet)) {
+            // Recursively search members and candidates (if EntitySet is a CandidateSet) for all PhysicalEntities.
+            physicalEntities.addAll(findEntitySetPhysicalEntities(multiPEInstance));
+        }
+        return physicalEntities;
```

Comment:
The logic for finding physical entities in a complex and polymer is almost identical except for the attribute value of "hasComponent" and "repeatedUnit".  You could consolidate it into one method that takes the attribute to use to get the constituent as a parameter.  It would also simplify the body of this method so it's clear it's only switching based on whether the instance is a complex, polymer, or set.

Reply:
I joined them in a helper method. I considered using a ternary in the method:

`return complexOrPolymerInst.getSchemClass().isa("Complex") ? complexOrPolymerInst.getAttributeValuesList("hasComponent") : complexOrPolymerInst.getAttributeValuesList("repeatedUnit")`

But I decided to use the less concise if/elses because the ternary made Polymers more of an afterthought to the method